application/src/test/java/org/thingsboard/server/transport/coap/attributes/AbstractCoapAttributesIntegrationTest.java

```diff
@@ -235,25 +235,37 @@ protected void processJsonTestSubscribeToAttributesUpdates(boolean emptyCurrentS
         CoapTestCallback callbackCoap = new CoapTestCallback(1);
 
         CoapObserveRelation observeRelation = client.getObserveRelation(callbackCoap);
-        callbackCoap.getLatch().await(3, TimeUnit.SECONDS);
-
+        String awaitAlias = "await Json Test Subscribe To AttributesUpdates (client.getObserveRelation)";
+        await(awaitAlias)
+                .atMost(10, TimeUnit.SECONDS)
+                .until(() -> CoAP.ResponseCode.CONTENT.equals(callbackCoap.getResponseCode()) &&
+                        callbackCoap.getObserve() != null &&
+                        0 == callbackCoap.getObserve().intValue());
         if (emptyCurrentStateNotification) {
-            validateUpdateAttributesJsonResponse(callbackCoap, "{}", 0);
+            validateUpdateAttributesJsonResponse(callbackCoap, "{}");
         } else {
-            validateUpdateAttributesJsonResponse(callbackCoap, SHARED_ATTRIBUTES_PAYLOAD_ON_CURRENT_STATE_NOTIFICATION, 0);
+            validateUpdateAttributesJsonResponse(callbackCoap, SHARED_ATTRIBUTES_PAYLOAD_ON_CURRENT_STATE_NOTIFICATION);
         }
 
-        CountDownLatch latch = new CountDownLatch(1);
-        int expectedObserveCnt = callbackCoap.getObserve().intValue() + 1;
+        int expectedObserveBeforeAddCnt = callbackCoap.getObserve().intValue() + 1;
         doPostAsync("/api/plugins/telemetry/DEVICE/" + savedDevice.getId().getId() + "/attributes/SHARED_SCOPE", SHARED_ATTRIBUTES_PAYLOAD, String.class, status().isOk());
-        latch.await(3, TimeUnit.SECONDS);
-        validateUpdateAttributesJsonResponse(callbackCoap, SHARED_ATTRIBUTES_PAYLOAD, expectedObserveCnt);
+        awaitAlias = "await Json Test Subscribe To AttributesUpdates (add attributes)";
+        await(awaitAlias)
+                .atMost(10, TimeUnit.SECONDS)
+                .until(() -> CoAP.ResponseCode.CONTENT.equals(callbackCoap.getResponseCode()) &&
+                        callbackCoap.getObserve() != null &&
+                        expectedObserveBeforeAddCnt == callbackCoap.getObserve().intValue());
+        validateUpdateAttributesJsonResponse(callbackCoap, SHARED_ATTRIBUTES_PAYLOAD);
 
-        latch = new CountDownLatch(1);
         int expectedObserveBeforeDeleteCnt = callbackCoap.getObserve().intValue() + 1;
```

Comment:
Hi @nickAS21 , please change the name of this property to expectedObserveAfterAttributesDelete

Reply:
please fix the naming in other places as well.